Use mocked scm client in development and add dummy data generator

JoachimTislov opened the pull request:
Closes: #1341

JoachimTislov committed:
feat: Create initial dummy data generator

meling commented:
Just for the record. I support developing this. It is not my own priority, but I appreciate your efforts on this @JoachimTislov. One small comment about naming; I suggest calling the internal package `mockdata` and the command `cmd/mockgen` or `cmd/mockdatagen`. The only issue with `mockgen` is that there's another tool with the same name. But that shouldn't be a problem if we only run it from the `Makefile` using `go run`. The make target could be `make mockdata`, which reads nicely.

JoachimTislov committed:
refactor: rename dummydata to mockdata package

- Restructure mock data
- Add support for custom "mock.json" and handle fallback to default
  values, if the file does not exist
- Add configurable parameters, and enhance logging.
  - Add verbose logging to prevent spam in terminal
- Rename cmd/dummyDB to cmd/mockgen
- Makefile target from dev-db to mockdata.

JoachimTislov committed:
refactor(mockdata): Add getters and export config fields
 - Resolve golangci-lint issues

JoachimTislov committed:
feature(mockdata): Add support for custom configuration

JoachimTislov committed:
fix: golangci-lint issues
